File: backend/internal/models/comments.go
package models

type Comment struct {
	ID              int    `json:"id"`
	Content         string `json:"content"`
	PostID          int    `json:"postId"`
	PostTitle       string `json:"postTitle"`
	CreatorUsername string `json:"creatorUsername"`
	CreatedAt       string `json:"createdAt"`
}

File: frontend/src/api/comments.ts
import { apiFetch } from "./client";
import type { Comment } from "../types/comment";

export function getComments(postId: number): Promise<Comment[]> {
  return apiFetch(`/api/comments?postId=${postId}`);
}

export function createComment(content: string, postId: number) {
  return apiFetch("/api/comments", {
    method: "POST",
    body: JSON.stringify({ content, postId }),
  });
}

export function deleteComment(id: number) {
  return apiFetch(`/api/comments/${id}`, { method: "DELETE" });
}
